### ISSUE TITLE
[blog post] On the Rule of Three and Cute Code

### ISSUE THREAD

**Issue by dev-cprice** [dev-cprice/codyaprice.com#81]
Dan Abramov post
[Goodbye, Clean Code — Overreacted](https://overreacted.io/goodbye-clean-code/)

@getify response, (mis)abstraction
[Goodbye, Clean Code - link - javascript - Reddit](https://www.reddit.com/r/javascript/comments/enizl1/goodbye_clean_code/fe2fou8/)

Rule of Three
[Rule of three (computer programming) - Wikipedia](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))

DRY
[Don’t repeat yourself - Wikipedia](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

AHA, by Kent C Dodds
[AHA Programming 💡](https://kentcdodds.com/blog/aha-programming)

https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction

**Comment by dev-cprice** [comment]
https://twitter.com/codewisdom/status/1229465545315147782?s=21

**Comment by dev-cprice** [comment]
https://github.com/dwmkerr/hacker-laws#kernighans-law

**Comment by dev-cprice** [comment]
http://nomodes.com/Larry_Tesler_Consulting/Complexity_Law.html

**Comment by getify** [comment]
On that "complexity law", I agree with most of what's said there, except one little tiny piece. The use of "complexity" throughout is inappropriate, IMO. s/complexity/difficulty. Complexity is a different beast entirely, and doesn't hold the meaning this law is targeting.

**Comment by dev-cprice** [comment]
Yeah, that one was going to be more or less _"loosely interpreted"_. Mainly, I want to drive home the point:

> Every application has an inherent amount of irreducible complexity

which, to me, implies that you should design your code _"well"_ (here comes SRP, etc etc). A side-effect of this is that sometimes the problem space is complicated, but that doesn't mean you need to make it any more complicated by writing convoluted code.
